repo-updater: Implement GitLab exclude feature (#3014)
sourcegraph/sourcegraph: CHANGELOG.md

- Enterprise admins can now customize the appearance of the homepage and search icon.
- A new settings property `notices` allows showing custom informational messages on the homepage and at the top of each page.

> - The `gitlab.exclude` setting was added to the [GitLab external service config](https://docs.sourcegraph.com/admin/external_service/gitlab#configuration) to allow excluding repositories yielded by `gitlab.projectQuery` from being synced.

Reviewer: ```suggestion
- The new `gitlab.exclude` setting in [GitLab external service config](https://docs.sourcegraph.com/admin/external_service/gitlab#configuration) allows you to exclude specific repositories matched by `gitlab.projectQuery` (so that they won't be synced).
```
